uikit/common/src/main/java/com/sygic/ui/common/views/BottomSheetDialog.kt

```diff
@@ -0,0 +1,190 @@
+package com.sygic.ui.common.views
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import android.util.TypedValue
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams
+import android.view.Window
+import android.view.WindowManager
+import android.widget.FrameLayout
+import androidx.annotation.LayoutRes
+import androidx.annotation.StyleRes
+import androidx.appcompat.app.AppCompatDialog
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.core.view.AccessibilityDelegateCompat
+import androidx.core.view.ViewCompat
+import androidx.core.view.accessibility.AccessibilityNodeInfoCompat
+import com.google.android.material.R.*
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+
+/**
+ * Friendly and more usable alternative to the official material BottomSheetDialog
+ */
+class BottomSheetDialog @JvmOverloads constructor(
+    context: Context,
+    @StyleRes theme: Int,
+    private val initialPeekHeight: Int? = null,
+    private val initialState: Int = BottomSheetBehavior.STATE_COLLAPSED
+) : AppCompatDialog(context, getSubThemeResId(context, theme)) {
+
+    var behavior: BottomSheetBehavior<FrameLayout>? = null
```

Comment:
toto by mohlo byt aj lateinit nie? Vzdy musi nejaky existovat, nikdy nebude null "po inicializacii"

Reply:
je to public a inicializuje se až v té metodě wrapInBottomSheet, raději bych to nechal nullable